ceph/cephfs/cephfs-provisioner.go

```diff
@@ -284,6 +288,73 @@ func (p *cephFSProvisioner) Delete(volume *v1.PersistentVolume) error {
 	return nil
 }
 
+// Look up the cluster dns service by label "coredns", falling back to "kube-dns" if not found
+func findDNSIP(p *cephFSProvisioner) (dnsip string) {
```

Comment:
how about moving these functions into a util package, then we can don't have duplicate code both in rbd/cephfs provisioners? https://github.com/kubernetes-incubator/external-storage/blob/master/ceph/rbd/pkg/provision/provision.go#L193

Reply:
@cofyc I could do that. I would suggest to only move ``lookuphost`` (should be renamed to ``lookupHost`` I think), ``splitHostPort`` and ``joinHostPort`` there as findDNSIP takes a provisioner specific struct and would require more changes. What would be the right file to put this? https://github.com/kubernetes-incubator/external-storage/blob/master/lib/util/util.go?

Reply:
From my point, `findDNSIP` can take one argument `client` to get DNS IP, e.g.

```
if p.dnsip == "" {
    p.dnsip = util.FindDNSIP(p.client)
}
```

lib/util/util.go is right file to put these functions.

Reply:
Thank you for the hints. I think I've addressed your suggestions.